[docs-only] Update README for invitations (#5915)
owncloud/ocis: services/invitations/README.md

The corresponding CS3 API [user types](https://cs3org.github.io/cs3apis/#cs3.identity.user.v1beta1.UserType) used to reperesent this are: `USER_TYPE_GUEST` and `USER_TYPE_PRIMARY`.

## Subcommand

This service introduces the `invitations` subcommand to `ocis`.

> ## Provisioning backends

Reviewer: ```suggestion
## Provisioning Backends
```

  "directGrantFlow": "direct grant",
  "resetCredentialsFlow": "reset credentials",
  "clientAuthenticationFlow": "clients",
  "dockerAuthenticationFlow": "docker auth",
  "attributes": {
    "cibaBackchannelTokenDeliveryMode": "poll",
    "cibaAuthRequestedUserHint": "login_hint",
    "oauth2DevicePollingInterval": "5",
    "clientOfflineSessionMaxLifespan": "0",
    "clientSessionIdleTimeout": "0",
    "clientOfflineSessionIdleTimeout": "0",
    "cibaInterval": "5",
    "realmReusableOtpCode": "false",
    "cibaExpiresIn": "120",
    "oauth2DeviceCodeLifespan": "600",
    "parRequestUriLifespan": "60",
    "clientSessionMaxLifespan": "0",
    "adminEventsExpiration": ""
  },
  "keycloakVersion": "21.0.1",
  "userManagedAccessAllowed": false,
  "clientProfiles": {
    "profiles": []
  },
  "clientPolicies": {
    "policies": []
  }
}
```

Importing this into keycloak will give you a realm that federates with an LDAP server, has the right
clients configured, and all the mappers correctly set. Be sure to set all the credentials after the import,
as they will be disabled.

The most relevant bits here are the mappers for the `OWNCLOUD_ID` and `OWNCLOUD_USER_TYPE` user properties.

> #### Backend configuration

Reviewer: ```suggestion
#### Backend Configuration
```

> After keycloak has been configured, you need to configure the backend with the following environment variables:

Reviewer: ```suggestion
After Keycloak has been configured, the invitation service needs to be configured with the following environment variables:
```

> * `INVITATIONS_KEYCLOAK_BASE_PATH`: The URL on which Keycloak is available.
> * `INVITATIONS_KEYCLOAK_CLIENT_ID`: The client ID of the client to use, in the above example, this would be `invitations`.
> * `INVITATIONS_KEYCLOAK_CLIENT_SECRET`: The client secret to use to authenticate, you can find this in the keycloak UI.
> * `INVITATIONS_KEYCLOAK_CLIENT_REALM`: The realm where the client was added, in the example above it would be `ocis`
> * `INVITATIONS_KEYCLOAK_USER_REALM`: The realm where to add the users, in the example above it would also be `ocis`.
> * `INVITATIONS_KEYCLOAK_INSECURE_SKIP_VERIFY`: If set to true, it will skip the verification of the keycloak https certificate. Not recommended in production enviroments.

Reviewer: ```suggestion
* `INVITATIONS_KEYCLOAK_BASE_PATH`: The URL to access Keycloak.
* `INVITATIONS_KEYCLOAK_CLIENT_ID`: The client ID of the client to use. In the above example, `invitations` is used.
* `INVITATIONS_KEYCLOAK_CLIENT_SECRET`: The client secret used to authenticate. This can be found in the Keycloak UI.
* `INVITATIONS_KEYCLOAK_CLIENT_REALM`: The realm where the client was added. In the example above, `ocis` is used.
* `INVITATIONS_KEYCLOAK_USER_REALM`: The realm where to add the users. In the example above, `ocis` is used.
* `INVITATIONS_KEYCLOAK_INSECURE_SKIP_VERIFY`: If set to true, the verification of the Keycloak https certificate is skipped. This is not recommended in production enviroments.
```

The corresponding CS3 API [user types](https://cs3org.github.io/cs3apis/#cs3.identity.user.v1beta1.UserType) used to reperesent this are: `USER_TYPE_GUEST` and `USER_TYPE_PRIMARY`.

## Subcommand

> This service introduces the `invitations` subcommand to `ocis`.

Reviewer: Q: with subcommand you mean a CLI? In case we need to rephrase the stuff a bit and propose an example, see: https://doc.owncloud.com/ocis/next/deployment/services/s-list/storage-users.html#manage-unfinished-uploads

## Provisioning backends

### Keycloak

> The default (and currently only) backend used to handle Invitations is [Keycloak](https://www.keycloak.org/).
> Keycloak is an open source identity and access management (IAM) system that other OCIS services also support as an
> authentication and authorization backend.
> 

Reviewer: ```suggestion
The default and currently only available backend used to handle Invitations is [Keycloak](https://www.keycloak.org/). Keycloak is an open source identity and access management (IAM) system which is also integrated by other OCIS services as an authentication and authorization backend.
```

#### Realm configuration

> This is the configuration of the Keycloak realm that this backend has been configured with, this includes
> the configuration of the client (invitations):

Reviewer: ```suggestion
This is the configuration of the Keycloak realm that this backend has been configured with and includes the client configuration for invitations:
```
Questions:
* to use this json, are there any prerequisites that need to be met on the KeyCloak side?
* the json is pretty large making the document hard to read.
I am thinking about to have that as linked file. Like in a folder named `md-sources` (can be re-used for images too).
This directory name is currently not used in any service.

Author: 1. No.
2. That's not a bad idea.
